app/controllers/shelter_pets_controller.rb

```diff
@@ -1,6 +1,6 @@
 class ShelterPetsController < ApplicationController
 
   def index
-    @pets = Pet.all
+   @shelter = Shelter.find(params[:shelter_id])
```

Comment:
This fixed the bug where each shelter was displaying all pets. We need to give the index access to only one shelter, not all of them.